From c3e2303af8c46e554715a06c980be50ba11024f9 Mon Sep 17 00:00:00 2001
From: Joel Lopes Da Silva <joel@lopes-da-silva.com>
Date: Mon, 21 Jan 2019 14:19:17 -0800
Subject: [PATCH] figlet: Enable a compiler option that allows figlet to query
 the width of the terminal window, so it will be able to print centered
 messages.

---
 Formula/figlet.rb | 143 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 143 insertions(+)

diff --git a/Formula/figlet.rb b/Formula/figlet.rb
index 77af2a223675..94676596881f 100644
--- a/Formula/figlet.rb
+++ b/Formula/figlet.rb
@@ -28,6 +28,8 @@ class Figlet < Formula
     sha256 "e6493f51c96f8671c29ab879a533c50b31ade681acfb59e50bae6b765e70c65a"
   end
 
+  patch :DATA
+
   def install
     (pkgshare/"fonts").install resource("contrib"), resource("intl")
 
@@ -45,3 +47,144 @@ class Figlet < Formula
     system "#{bin}/figlet", "-f", "larry3d", "hello, figlet"
   end
 end
+__END__
+diff a/Makefile b/Makefile
+--- a/Makefile
++++ b/Makefile
+@@ -19,14 +19,16 @@
+ SHELL = /bin/sh
+ 
+ # The C compiler and linker to use
+-CC	= gcc
+-CFLAGS	= -g -O2 -Wall
+-LD	= gcc
+-LDFLAGS =
++CC	?= clang
++CFLAGS	+= -g -Wall
++LD	?= clang
++LDFLAGS	+= -macosx_version_min 10.13 -lc
+ 
+ # Feature flags:
+ #   define TLF_FONTS to use TOIlet TLF fonts
+ XCFLAGS	= -DTLF_FONTS
++#   OS X is unix
++XCFLAGS	+= -Dunix
+ 
+ # Where to install files
+ prefix	= /usr/local
+diff a/figlet.c b/figlet.c
+--- a/figlet.c
++++ b/figlet.c
+@@ -358,7 +358,7 @@
+     if (dummy == '\n') return;
+     if (dummy == '\r') {
+       dummy = Zgetc(fp);
+-      if (dummy != EOF && dummy != '\n') Zungetc(dummy,fp);
++      if (dummy != EOF && dummy != '\n') (void)Zungetc(dummy,fp);
+       return;
+       }
+   }
+@@ -388,7 +388,7 @@
+     if (c=='\n') break;
+     if (c=='\r') {
+       c = Zgetc(fp);
+-      if (c != EOF && c != '\n') Zungetc(c,fp);
++      if (c != EOF && c != '\n') (void)Zungetc(c,fp);
+       *(p-1) = '\n';
+       break;
+       }
+@@ -498,7 +498,7 @@
+ {
+   int c;
+   while (c=Zgetc(fp),isascii(c)&&isspace(c)) ;
+-  Zungetc(c,fp);
++  (void)Zungetc(c,fp);
+   }
+ 
+ /****************************************************************************
+@@ -526,7 +526,7 @@
+     sign = -1;
+     }
+   else {
+-    Zungetc(c,fp);
++    (void)Zungetc(c,fp);
+     }
+   c = Zgetc(fp);
+   if (c=='0') {
+@@ -536,19 +536,19 @@
+        }
+      else {
+        base = 8;
+-       Zungetc(c,fp);
++       (void)Zungetc(c,fp);
+        }
+     }
+   else {
+     base = 10;
+-    Zungetc(c,fp);
++    (void)Zungetc(c,fp);
+     }
+ 
+   while((c=Zgetc(fp))!=EOF) {
+     c=toupper(c);
+     p=strchr(digits,c);
+     if (!p) {
+-      Zungetc(c,fp);
++      (void)Zungetc(c,fp);
+       *nump = acc * sign;
+       return;
+       }
+@@ -576,7 +576,7 @@
+ 
+   thechar=Zgetc(fp);
+   if (thechar=='\n' || thechar=='\r') { /* Handle badly-formatted file */
+-    Zungetc(thechar,fp);
++    (void)Zungetc(thechar,fp);
+     return '\0';
+     }
+   if (thechar!='\\') return thechar;
+@@ -600,7 +600,7 @@
+       return 11;
+     default:
+       if (next=='-' || next=='x' || (next>='0' && next<='9')) {
+-        Zungetc(next,fp);
++        (void)Zungetc(next,fp);
+         readnum(fp,&thechar);
+         return thechar;
+         }
+@@ -625,7 +625,7 @@
+   result = readTchar(fp);
+   if (result == '\n' || result == '\r') {
+     result = 0;
+-    Zungetc(result,fp);
++    (void)Zungetc(result,fp);
+     }
+   return result;
+   }
+@@ -677,7 +677,7 @@
+      skiptoeol(controlfile);
+      return;
+      }
+-  Zungetc(ch, controlfile);
++  (void)Zungetc(ch, controlfile);
+   skipws(controlfile);
+   gn[n] = 65536L * charsetname(controlfile);
+   gndbl[n] = 0;
+@@ -766,7 +766,7 @@
+           lastch=readTchar(controlfile);
+           }
+         else {
+-          Zungetc(dashcheck,controlfile);
++          (void)Zungetc(dashcheck,controlfile);
+           lastch=firstch;
+           }
+         skipws(controlfile);
+@@ -784,7 +784,7 @@
+       case '5': case '6': case '7': case '8': case '9':
+       case '-':
+                 /* Mapping table entry */
+-        Zungetc(command,controlfile);
++        (void)Zungetc(command,controlfile);
+         readnum(controlfile,&firstch);
+         skipws(controlfile);
+ 	readnum(controlfile,&lastch);
-- 
2.20.1

